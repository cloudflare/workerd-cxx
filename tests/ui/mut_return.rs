#[cxx::bridge]
mod ffi {
    extern "Rust" {
        type Mut<'a>;
    }

    unsafe extern "C++" {
        type Thing;

        fn f<'a>(t: &'a Thing) -> Pin<&'a mut CxxString>;
        unsafe fn g<'a>(t: &'a Thing) -> Pin<&'a mut CxxString>;
        fn h<'a>(t: Box<Mut>) -> Pin<&'a mut CxxString>;
        fn i<'a>(t: Box<Mut<'a>>) -> Pin<&'a mut CxxString>;
        fn j<'a>(t: &'a Thing) -> &'a mut [u8];
    }
}

fn main() {}
