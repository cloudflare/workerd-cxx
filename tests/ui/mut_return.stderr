error: &mut return type is not allowed unless there is a &mut argument
  --> tests/ui/mut_return.rs:10:9
   |
10 |         fn f<'a>(t: &'a Thing) -> Pin<&'a mut CxxString>;
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: &mut return type is not allowed unless there is a &mut argument
  --> tests/ui/mut_return.rs:14:9
   |
14 |         fn j<'a>(t: &'a Thing) -> &'a mut [u8];
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
